apiary.apib

```diff
@@ -9755,3 +5182,343 @@ Retrieves projects dependency graph for a user, and optionally for a specific pr
++ `allowedOperations` (array, required)
+    + (enum[string])
+        + Members
+            + `ChangeProjectSettings`
+            + `Run`
+            + `ConsumeApiEndpoint`
+            + `Edit`
+            + `ViewWorkspaces`
+            + `UpdateProjectDescription`
+            + `ProjectSearchPreview`
+            + `BrowseReadFiles`
+            + `EditTags`
+            + `RunLauncher`
+            + `ViewRuns`
++ `runsCountByType` (array[domino.server.projects.api.ProjectGatewayExecutingRunsByType], required)
++ `environmentName` (string, required)
++ `name` (string, required)
++ `serviceProvider` (enum[string], optional, nullable)
+    + `github`
+    + `gitlab`
+    + `githubEnterprise`
+    + `unknown`
+    + `gitlabEnterprise`
+    + `bitbucket`
+    + `bitbucketServer`
++ `id` (string, required)
++ `updatedAt` (string, required)
++ `stageId` (string, required)
++ `status` (domino.server.projects.api.ProjectStatus, required)
+
+## domino.projects.api.ProjectStatus (object)
+
+
+### Properties
++ `status` (enum[string], required)
+    + `active`
+    + `complete`
++ `isBlocked` (boolean, required)
++ `blockedReason` (string, optional, nullable)
++ `completedMessage` (string, optional, nullable)
+
+
+## domino.server.projects.api.ProjectStatus (object)
+
+
+### Properties
++ `status` (enum[string], required)
+    + `active`
+    + `complete`
++ `isBlocked` (boolean, required)
++ `blockedReason` (string, optional, nullable)
++ `completedMessage` (string, optional, nullable)
+
+## domino.server.projects.api.ProjectGatewayExecutingRunsByType (object)
+
+
+### Properties
++ `runType` (enum[string], required)
+    + `Other`
+    + `SSHProxy`
+    + `Endpoint`
+    + `App`
+    + `Scheduled`
+    + `Launcher`
+    + `Batch`
+    + `Workspace`
++ `count` (number, required)
+
+## domino.repoman.domain.GitRepository (object)
+
+
+### Properties
++ `id` (string, required)
++ `name` (string, required)
++ `uriHost` (string, required)
++ `uriPort` (string, optional, nullable)
++ `uriPath` (string, required)
++ `serviceProvider` (enum[string], required)
+    + `github`
+    + `gitlab`
+    + `githubEnterprise`
+    + `unknown`
+    + `gitlabEnterprise`
+    + `bitbucket`
+    + `bitbucketServer`
+
+
+## domino.server.projects.api.ProjectGatewayTag (object)
+
+
+### Properties
++ `id` (string, required)
++ `name` (string, required)
++ `isApproved` (boolean, required)
+
+## domino.server.projects.api.Owner (object)
+
+
+### Properties
++ `id` (string, optional, nullable)
++ `userName` (string, required)
+
+## domino.notifications.UpdateNotificationRequest (object)
+
+
+### Properties
++ `expired` (boolean, optional) - mark the notification as expired (only true is supported)
+
+
+## domino.notifications.ListAdminNotificationsResult (object)
+
+
+### Properties
++ `pagination` (domino.notifications.Pagination, optional)
++ `notifications` (array[domino.notifications.AdminNotification], optional)
```

Comment:
neither of these are optional. will have to update that in swagger

Reply:
this and all comments below refer to objects returned from the server, so the question whether st is optional is not that important, however something being non-optional means client code doesn't have to be all that defensive